delta_filter.cpp

```diff
@@ -82,10 +83,22 @@ void DeltaFilter::ingest(vector<Reading *> *readings, vector<Reading *>& out)
 			m_state.insert(pair<string, DeltaData *>(delta->getAssetName(), delta));
 			out.push_back(*it);
 		}
-		else if (deltaIt->second->evaluate(reading, getTolerance(reading->getAssetName()), m_rate))
+		else if (deltaIt->second->evaluate(reading, m_toleranceMeasure, getTolerance(reading->getAssetName()), m_rate, 
+                                            m_processingMode, sendOrig, readingToSend))
 		{
-			// This reading needs to be sent onwards
-			out.push_back(*it);
+            // evaluate's return value indicates whether a reading needs to be sent onwards
+            if(sendOrig)
+            {
+                // out.push_back(move(*it)); // TODO: check if it is possible to just move the reading ptr to out vector
+                // *it = nullptr;
+                out.push_back(new Reading(*reading));
+                delete *it;
```

Comment:
can be moved out of if else block.

Reply:
"sendOrig == true" case can probably be optimized, hence want to keep it separate logically.